Lib/unittest/mock.py

```diff
@@ -1601,7 +1601,11 @@ def _patch_object(
     When used as a class decorator `patch.object` honours `patch.TEST_PREFIX`
     for choosing which methods to wrap.
     """
-    getter = lambda: target
+    if type(target) is str:
+        raise TypeError(f"Target {target} must be a class")
```

Comment:
target does not need to be a class, it can be an instance of a class, a module or anything else patcheable. Example:
```
>>> import sys
>>> patch.object(sys, "executable").start()
<MagicMock name='executable' id='720575940383700296'>
>>> sys.executable
<MagicMock name='executable' id='720575940383700296'>
```

I'd rather say something like:

`raise TypeError(f"{target!r} must be the actual object to patch, not a `str`")`

That is "more correct" and tells the user what to do.

Reply:
done